### ISSUE TITLE
$page props in child components cause routing failure

### ISSUE THREAD

**Issue by parker-codes** [inertiajs/inertia#177]
I'm running into a hard-to-track issue where any usage of $page in components that are not at the top page level cause routing to malfunction. I've noticed that the error happens when conditionally displayed components are involved.

In my case, I have a Coupons page. It has a NewCouponForm component that is conditionally rendered based on if a button is clicked to create a new coupon. Inside that NewCouponForm component, I am accessing the $page store to get the `type` property.

`Coupons.svelte`:

```
<Layout title="Coupons">
  <div class="h-screen">
    <div class="text-center mt-6">
      <h2 class="text-4xl tracking-wide text-gray-800">Coupons</h2>
    </div>

    <div class="w-1/2 mx-auto mt-6">
      {#if showNewCouponForm}
        <NewCouponForm on:created={addNewCoupon} />
      {:else}
        <div class="flex justify-end">
          <button
            on:click|preventDefault={() => (showNewCouponForm = true)}
            class="bg-teal-500 text-white font-semibold hover:text-teal-700
            hover:bg-transparent py-1 px-2 border border-teal-500 rounded">
            Add New Coupon
          </button>
        </div>
      {/if}
    </div>

    <div class="w-1/2 mx-auto mt-8">

      {#each coupons as coupon}
        <div
          on:click={() => (selectedCouponId = coupon.id)}
          class="mt-8 first:mt-0 cursor-pointer">
          <Coupon
            bind:coupon
            {types}
            selected={selectedCouponId === coupon.id} />
        </div>
      {/each}

    </div>
  </div>
</Layout>
```

The `NewCouponForm.svelte` (child component) is fairly large, but it has this snippet:

```
<select
  bind:value={coupon.type}
  name="type"
  required
  id="coupon-{coupon.id}-type">
  {#each $page.types as type}
    <option value={type.key}>{type.description}</option>
  {/each}
</select>
```

As you can see, this NewCouponForm has a reference to $page. The page actually renders this just fine, and I like being able to access the global page store without having to pass props down. 

The issue shows up when I try to navigate to a different page once I've clicked the button to create a new form. Since the NewCouponForm component was displayed, it accessed the $page.types value to use it in a dropdown. When I click on an InertiaLink in the navbar, the page does not transition. No console errors either.

If I visit the Coupons page but do NOT click on the button that shows the NewCouponForm, then I can navigate away just fine.

Once I changed my code to pass down the `types` as a prop, there were no issues. This isn't desired though since the store is supposed to be globally accessible.

---

I did my best to cover what I know about the issue, but it may not be expansive. Just let me know what questions you have. :)

**Comment by reinink** [comment]
@pedroborges Do you know what's up here, or this is still an issue?

**Comment by reinink** [comment]
Okay, so I think this issue is related to #476 and fixed in #522. 

@parker-codes You still trying to solve this? 😅

**Comment by parker-codes** [comment]
@reinink No. I use Adonis for projects now and haven't been able to get a good Inertia setup unfortunately, so this issue was on a project I haven't touched in a long time. Definitely still a Svelte user though!

**Comment by reinink** [comment]
@parker-codes Understood! I think that this has been fixed in `@inertiajs/inertia-svelte@0.6.1`, if you ever want to try again. Feel free to report if it's not. Going to close this issue in the meantime. 👍
